src/renderer/fullscreenQuad.js

```
@@ -18,8 +18,8 @@ export function makeFullscreenQuad(gl) {
     gl.drawArrays(gl.TRIANGLES, 0, 6);
   }
 
-  return Object.freeze({
+  return {
     draw,
```

Usually, the best way to treat immutables is a pattern that utilizes proxies and getter and setters. Basically you pass the initial data, and a Proxy that just allows you to access that data is created. You can even throw exceptions if someone tries to use data outside of the initial object structure. That pattern works well in these scenarios!

Let me know if you'd like me to expand or jump in a call to work it out together. That's the tiny bit I can add to this project :sweat_smile:

---

I'd like to learn more! Do I get what you're talking about? You return `get`ters and `set`ters to properties instead of the properties themselves, or otherwise return functions that control access to the module instead of returning the mutable state directly? If that's so, how do you return the module object itself so that it's not accidentally mutated? That's why I was using `Object.freeze` before, but that doesn't seem like a great solution, since it silently fails.